tests/unit/etlng/SchedulingTests.cpp

```diff
@@ -0,0 +1,187 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of clio: https://github.com/XRPLF/clio
+    Copyright (c) 2025, the clio developers.
+
+    Permission to use, copy, modify, and distribute this software for any
+    purpose with or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include "etlng/Models.hpp"
+#include "etlng/SchedulerInterface.hpp"
+#include "etlng/impl/Loading.hpp"
+#include "etlng/impl/Scheduling.hpp"
+#include "util/LoggerFixtures.hpp"
+#include "util/MockNetworkValidatedLedgers.hpp"
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <functional>
+#include <memory>
+#include <optional>
+#include <utility>
+
+using namespace etlng;
+using namespace etlng::model;
+
+namespace {
+class FakeScheduler : SchedulerInterface {
+    std::function<std::optional<Task>()> generator_;
+
+public:
+    FakeScheduler(std::function<std::optional<Task>()> generator) : generator_{std::move(generator)}
+    {
+    }
+
+    [[nodiscard]] std::optional<Task>
+    next() override
+    {
+        return generator_();
+    }
+};
+}  // namespace
+
+struct ForwardSchedulerTests : NoLoggerFixture {
+protected:
+    MockNetworkValidatedLedgersPtr networkValidatedLedgers_;
+};
+
+TEST_F(ForwardSchedulerTests, ExhaustsSchedulerIfMostRecentLedgerIsNewerThanRequestedSequence)
+{
+    auto scheduler = impl::ForwardScheduler(networkValidatedLedgers_, 0u, 10u);
+    EXPECT_CALL(*networkValidatedLedgers_, getMostRecent()).Times(11).WillRepeatedly(testing::Return(11u));
+
+    for (auto i = 0u; i < 10u; ++i) {
+        auto maybeTask = scheduler.next();
+
+        EXPECT_TRUE(maybeTask.has_value());
+        EXPECT_EQ(maybeTask->seq, i);
```

Comment:
There is no place to check the other parameter of `Task` - `priority`.

Reply:
Correct. That's for another PR - these smaller PRs are extracted out of a larger, more complete branch. Just so you don't have to review 20k lines at once 😃